### PR TITLE
feat: support passing twiml instances to twiml property

### PR DESCRIPTION
**Pull request by mattcole19**
# Fixes #

Feature to fix [537](https://github.com/twilio/twilio-node/issues/537).
Internal Link: [DI-2440](https://issues.corp.twilio.com/browse/DI-2440)

### CODE REVIEW

**Comment by childish-sambino** [comment]
Need to generate examples folder.

**Comment by mattcole19** [comment]
Changes to twilio-node: https://github.com/twilio/twilio-node/pull/879

File: src/main/java/com/twilio/oai/resolver/node/NodeParameterResolver.java [hunk 1]
```diff
@@ -27,6 +27,12 @@ public CodegenParameter resolve(final CodegenParameter codegenParameter) {
             codegenParameter.vendorExtensions.put("x-transform", transform);
         }
 
+        // Feature to allow desired methods to accept TwiMl objects
+        if (codegenParameter.paramName.equalsIgnoreCase("twiml")){
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/resolver/node/NodeParameterResolver.java:31]: Would it be better to move the configuration for the `twiml` type to the type config file, similar to what's done for Java and C#?

File: src/main/resources/twilio-node/imports.mustache [hunk 2]
```diff
@@ -7,3 +7,4 @@ import { {{key}} } from "{{apiVersionPath}}/../../../{{value}}";
 {{/vendorExtensions.x-import}}
 {{/vars}}
 {{/responseModel}}
+{{#hasTwimlType}}import TwiML from "{{apiVersionPath}}/../../twiml/TwiML";{{/hasTwimlType}}
```
Comment (childish-sambino) [inline src/main/resources/twilio-node/imports.mustache:10]: Can this be replaced with updating the `library` section of `node.json` to include an entry for `twiml`?

Comment (childish-sambino) [reply src/main/resources/twilio-node/imports.mustache:10]: Could we import them all once and then remove unused with tooling?

File: src/test/java/com/twilio/oai/TwilioGeneratorTest.java [hunk 3]
```diff
@@ -28,14 +28,14 @@
...
-                             Generator.TWILIO_GO,
+        return Arrays.asList(Generator.TWILIO_NODE);
+/*                             Generator.TWILIO_GO,
                              Generator.TWILIO_JAVA,
                              Generator.TWILIO_NODE,
                              Generator.TWILIO_PHP,
                              Generator.TWILIO_PYTHON,
                              Generator.TWILIO_RUBY,
-                             Generator.TWILIO_TERRAFORM);
+                             Generator.TWILIO_TERRAFORM)*/
```
Comment (childish-sambino) [inline src/test/java/com/twilio/oai/TwilioGeneratorTest.java:38]: Eep

File: src/main/java/com/twilio/oai/resolver/node/NodeParameterResolver.java [hunk 4]
```diff
@@ -27,6 +27,10 @@ public CodegenParameter resolve(final CodegenParameter codegenParameter) {
             codegenParameter.vendorExtensions.put("x-transform", transform);
         }
 
+        if (mapper.libraries().getMap(codegenParameter.dataFormat).isPresent()){
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/resolver/node/NodeParameterResolver.java:30]: nit: Could simplify to `ifPresent(imports -> ...`

File: src/main/resources/twilio-node/imports.mustache [hunk 5]
```diff
@@ -7,3 +7,13 @@ import { {{key}} } from "{{apiVersionPath}}/../../../{{value}}";
 {{/vendorExtensions.x-import}}
 {{/vars}}
 {{/responseModel}}
+{{#operations}}
+{{#allParams}}
```
Comment (childish-sambino) [inline src/main/resources/twilio-node/imports.mustache:11]: Not obvious to me, but how were you able to get rid of the duplicates?

**Review by childish-sambino — APPROVED** [review]

File: src/main/resources/twilio-node/imports.mustache (hunk 5 above)
Comment (mattcole19) [reply src/main/resources/twilio-node/imports.mustache:11]: The example only has it in a single operation, so it only appears in twilio-node. What tooling should we use to remove the extra? Should I add logic to [clean_unused_imports](https://github.com/twilio/twilio-oai-generator/blob/main/scripts/clean_unused_imports.py)?

Comment (childish-sambino) [reply src/main/resources/twilio-node/imports.mustache:11]: I'd prefer to use some off-the-shelf tooling like prettier, but doesn't look it supports removing duplicate imports. Given that, I think it's fine to add logic to the import cleaning script.

File: src/main/resources/twilio-node/imports.mustache (hunk 2 above)
Comment (mattcole19) [reply src/main/resources/twilio-node/imports.mustache]: Added [function](https://github.com/twilio/twilio-oai-generator/pull/391/files#diff-7e15ccacf5f960109accbd6a4cca462305184eb56f17bd54ff9de6bbb6ef8475R53) for it

File: src/main/resources/twilio-node/imports.mustache (hunk 5 above)
Comment (mattcole19) [reply src/main/resources/twilio-node/imports.mustache:11]: Added [function](https://github.com/twilio/twilio-oai-generator/pull/391/files#diff-7e15ccacf5f960109accbd6a4cca462305184eb56f17bd54ff9de6bbb6ef8475R53) for it

File: scripts/clean_unused_imports.py [hunk 6]
```diff
@@ -44,3 +48,26 @@ def remove_unused_imports(root_dir, language):
...
+def remove_duplicate_imports(root_dir, language):
+    for filename in locate(LANGUAGE_REGEX_MAP[language]["pattern"], root_dir):
+        with open(filename) as f:
+            all_lines = f.readlines()
+
+        # Find duplicate import statements
+        lines_to_remove = []
+        for n, line in enumerate(all_lines):
+            IMPORT_RE=LANGUAGE_REGEX_MAP[language]["regex"]
+            m = IMPORT_RE.match(line) # IMPORT FOUND!
```
Comment (childish-sambino) [inline scripts/clean_unused_imports.py:62]: nit: Should just name it `import_found`

**Comment by mattcole19** [comment]
[Generated changes](https://github.com/twilio/twilio-node/pull/879)

File: src/main/java/com/twilio/oai/resolver/node/NodeParameterResolver.java [hunk 7]
```diff
@@ -27,9 +27,7 @@ public CodegenParameter resolve(final CodegenParameter codegenParameter) {
             codegenParameter.vendorExtensions.put("x-transform", transform);
         }
 
-        if (mapper.libraries().getMap(codegenParameter.dataFormat).isPresent()){
-            codegenParameter.vendorExtensions.put("x-import", mapper.libraries().get(codegenParameter.dataFormat).orElseThrow());
-        }
+        mapper.libraries().getMap(codegenParameter.dataFormat).ifPresent(imports -> codegenParameter.vendorExtensions.put("x-import", mapper.libraries().get(codegenParameter.dataFormat).orElseThrow()));
```
Comment (childish-sambino) [inline src/main/java/com/twilio/oai/resolver/node/NodeParameterResolver.java:30]: The reason for switching to `ifPresent` is it gives you the value from the map. This means you don't have to `get` the value again. E.g., 
```ts
mapper.libraries().getMap(codegenParameter.dataFormat).ifPresent(imports -> codegenParameter.vendorExtensions.put("x-import", imports));
```

**Review by childish-sambino — APPROVED** [review]

**Pull request merged**